contracts/rewards/boosted-staking/BoostedTokenWrapper.sol

```diff
@@ -132,10 +132,18 @@ contract BoostedTokenWrapper is InitializableReentrancyGuard {
      * @param _amount Units of StakingToken
      */
     function _withdrawRaw(uint256 _amount) internal nonReentrant {
-        _rawBalances[msg.sender] -= _amount;
+        _reduceRaw(_amount);
         stakingToken.safeTransfer(msg.sender, _amount);
     }
 
+    /**
+     * @dev Reduced a given staked balance of sender (no transfer)
+     * @param _amount Units of StakingToken
+     */
+    function _reduceRaw(uint256 _amount) internal nonReentrant {
```

Comment:
adding this second `nonReentrant` will cause any calls to `_withdrawRaw` to fail